components/raftstore/src/store/peer.rs

```diff
@@ -3137,6 +3141,32 @@ where
                 };
             }
         }
+        let flags = WriteBatchFlags::from_bits_truncate(req.get_header().get_flags());
+        if flags.contains(WriteBatchFlags::STALE_READ) {
+            let read_ts =
+                decode_var_u64(&mut req.mut_header().take_flag_data().as_slice()).unwrap();
```

Comment:
Probably using fixed u64 is better. `ts` is usually too large to be encoded as varint efficiently.